File: 2026-08/src/day_29/prms_43165/Solution2.py
import sys

sys.setrecursionlimit(10**6)


def solution(numbers, target):
    n = len(numbers)

    def dfs(cur, depth):
        if depth == n:
            return int(cur == target)
        return dfs(cur + numbers[depth], depth + 1) + dfs(cur - numbers[depth], depth + 1)

    return dfs(0, 0)

File: 2026-08/src/day_29/prms_43165/Solution2.java
package day_29.prms_43165;

class Solution2 {
    public int solution(int[] numbers, int target) {
        return dfs(0, 0, numbers, target);
    }

    static int dfs(int cur, int depth, int[] numbers, int target) {
        if (depth == numbers.length) return cur == target ? 1 : 0;

        return dfs(cur + numbers[depth], depth + 1, numbers, target) +
                dfs(cur - numbers[depth], depth + 1, numbers, target);
    }
}
